lighthouse-core/report/v2/report-renderer.js

```diff
@@ -19,19 +19,190 @@
 
 window.ReportRenderer = class ReportRenderer {
   /**
-   * @param {!Object} report
+   * @param {Document} document
    */
-  constructor(report) {
-    this._report = report;
+  constructor(document) {
+    this._document = document;
   }
 
   /**
-   * @param {!Element} element
+   * @param {ReportJSON} report
+   * @return {Element}
    */
-  render(element) {
-    const pre = document.createElement('pre');
-    pre.textContent = JSON.stringify(this._report, null, 2);
-    element.appendChild(pre);
+  renderReport(report) {
+    try {
+      const element = this._renderReport(report);
+      const pre = this._document.createElement('pre');
+      pre.textContent = JSON.stringify(report, null, 2);
+      element.appendChild(pre);
+      return element;
+    } catch (e) {
+      return this._renderException(e);
+    }
+  }
+
+  /**
+   * @param {string} name
+   * @param {string=} className
+   * @returns {Element}
+   */
+  _createElement(name, className) {
+    const element = this._document.createElement(name);
+    if (className)
+      element.className = className;
+    return element;
+  }
+
+  /**
+   * @param {Element} parent
+   * @param {string} name
+   * @param {string=} className
+   * @returns {Element}
+   */
+  _createChild(parent, name, className) {
+    const child = this._createElement(name, className);
+    parent.appendChild(child);
+    return child;
+  }
+
+  /**
+   * @param {Element} parent
+   * @param {string} text
+   * @return {Node}
+   */
+  _createTextChild(parent, text) {
+    const child = this._document.createTextNode(text);
+    parent.appendChild(child);
+    return child;
+  }
+
+  /**
+   * @param {number} score
+   * @param {string} title
+   * @param {string} description
+   * @return {Element}
+   */
+  _renderScore(score, title, description) {
+    const element = this._createElement('div', 'lighthouse-score');
+    const value = this._createChild(element, 'div', 'lighthouse-score__value');
```

Comment:
could easily get away without having a `_createChild`:

`element.appendChild(this._createElement('div', 'lighthouse-score__value'));`